slsReceiverSoftware/src/slsReceiverTCPIPInterface.cpp

```diff
@@ -682,1580 +547,793 @@ int slsReceiverTCPIPInterface::set_roi(sls::ServerInterface2 &socket) {
-	readOutFlags retval = GET_READOUT_FLAGS;
-
-	// get args, return if socket crashed, ret is fail if receiver is not null
-	socket.receiveArg(ret, mess, &arg, sizeof(arg));
-	if(receiver == nullptr){
-		NullObjectError(ret, mess);
-		return FAIL;
-	}
+int slsReceiverTCPIPInterface::set_readout_flags(Interface &socket) {
+	auto arg = socket.receive<readOutFlags>();
 
 	if (myDetectorType == JUNGFRAU || myDetectorType == GOTTHARD || myDetectorType == MOENCH)
 		functionNotImplemented();
 
-	// base object not null
-	else if (ret == OK) {
-		// set
-		if (arg >= 0) {
-			// verify if receiver is unlocked and idle
-			if (VerifyLockAndIdle(ret, mess, fnum) == OK) {
-				FILE_LOG(logDEBUG1) << "Setting readout flag: " << arg;
-				ret = receiver->setReadOutFlags(arg);
-			}
-		}
-		// get
-		retval = receiver->getReadOutFlags();
-		validate((int)arg, (int)(retval & arg), std::string("set readout flags"), HEX);
-		FILE_LOG(logDEBUG1) << "Readout flags: " << retval;
+	if (arg >= 0) {
+		VerifyIdle(socket);
+		FILE_LOG(logDEBUG1) << "Setting readout flag: " << arg;
+		ret = impl()->setReadOutFlags(arg);
 	}
-	return socket.sendResult(true, ret, &retval, sizeof(retval), mess);
+	auto retval = impl()->getReadOutFlags();
+	validate((int)arg, (int)(retval & arg), std::string("set readout flags"), HEX);
+	FILE_LOG(logDEBUG1) << "Readout flags: " << retval;
+	return socket.sendResult(retval);
 }
 
-
-
-int slsReceiverTCPIPInterface::set_adc_mask(sls::ServerInterface2 &socket) {
-	ret = OK;
-	memset(mess, 0, sizeof(mess));
-	uint32_t arg = -1;
-	uint32_t retval = -1;
-
-	// get args, return if socket crashed, ret is fail if receiver is not null
-	socket.receiveArg(ret, mess, &arg, sizeof(arg));
-	if(receiver == nullptr){
-		NullObjectError(ret, mess);
-		return FAIL;
+int slsReceiverTCPIPInterface::set_adc_mask(Interface &socket) {
+	auto arg = socket.receive<uint32_t>();
+	VerifyIdle(socket);
+	FILE_LOG(logDEBUG1) << "Setting ADC enable mask: " << arg;
+	impl()->setADCEnableMask(arg);
+	auto retval = impl()->getADCEnableMask();
+	if (retval != arg) {
+		sprintf(mess, "Could not ADC enable mask. Set 0x%x, but read 0x%x\n", arg, retval);
+		throw RuntimeError(mess);
 	}
-
-	// base object not null
-	if (ret == OK) {
-		if (VerifyLockAndIdle(ret, mess, fnum) == OK) {
-			FILE_LOG(logDEBUG1) << "Setting ADC enable mask: " << arg;
-			receiver->setADCEnableMask(arg);
-		}
-		retval = receiver->getADCEnableMask();
-		if (ret == OK && retval != arg) {
-			ret = FAIL;
-			sprintf(mess, "Could not ADC enable mask. Set 0x%x, but read 0x%x\n", arg, retval);
-			FILE_LOG(logERROR) << mess;
-		}
-		FILE_LOG(logDEBUG1) << "ADC enable mask retval: " << retval;
-	}
-	return socket.sendResult(true, ret, &retval, sizeof(retval), mess);
+	FILE_LOG(logDEBUG1) << "ADC enable mask retval: " << retval;
+	return socket.sendResult(retval);
 }
 
-int slsReceiverTCPIPInterface::set_dbit_list(sls::ServerInterface2 &socket) {
-    ret = OK;
-    memset(mess, 0, sizeof(mess));
+int sls
```

Comment:
throw since it is a fail?

Reply:
I removed the check completely since we already check in the slsDetector

Reply:
Yes, you are right.